src/order/retrieve-order.ts

```diff
@@ -0,0 +1,68 @@
+import { fetch } from "undici";
+import config from "../config";
+import type { OrderResponseBody } from "@paypal/paypal-js";
+
+const {
+  paypal: { apiBaseUrl },
+} = config;
+
+type RetrieveOrderErrorResponse = {
+  [key: string]: unknown;
+  name: string;
+  message: string;
+  debug_id: string;
+};
+
+type HttpErrorResponse = {
+  statusCode?: number;
+  details?: Record<string, string>;
+} & Error;
+
+export async function retrieveOrder(accessToken: string, orderID: string) {
```

Comment:
nit: can we rename this to `getOrder` instead of `retrieveOrder`? That way we match the naming convention of the public v2 orders api.